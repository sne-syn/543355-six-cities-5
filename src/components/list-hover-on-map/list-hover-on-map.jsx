import React from 'react';
import {connect} from 'react-redux';
import {setActiveCardID, resetActiveCardID} from "../../store/action";
import List from '../list/list';

const ListHoverOnMap = (props) => {
  return (
    <List {...props}/>
  );
};

function mapStateToProps(state) {
  return {
    highlightedOfferID: state.highlightedOfferID,
    offers: state.filteredOffers
  };
}

const mapDispatchToProps = (dispatch) => ({
  setActiveCardID(evt) {
    dispatch(setActiveCardID(evt));
  },
  resetActiveCardID() {
    dispatch(resetActiveCardID());
  }
});

export {ListHoverOnMap};
export default connect(mapStateToProps, mapDispatchToProps)(ListHoverOnMap);
