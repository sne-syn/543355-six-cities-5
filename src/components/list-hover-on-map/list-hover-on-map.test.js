import React from 'react';
import renderer from 'react-test-renderer';
import {ListHoverOnMap} from './list-hover-on-map';
jest.mock(`../list/list`, () => `List`);

const noop = () => {};
const offers = [];

test(`ListHoverOnMap render`, () => {
  const tree = renderer
  .create(
      <ListHoverOnMap
        highlightedOfferID={`5`}
        type={`cities`}
        offers={offers}
        resetActiveCardIdAction={noop}
        setActiveCardIdAction={noop}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
