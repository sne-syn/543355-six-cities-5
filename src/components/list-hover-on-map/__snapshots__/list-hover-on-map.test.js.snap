// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHoverOnMap render 1`] = `
<List
  highlightedOfferID="5"
  offers={Array []}
  resetActiveCardIdAction={[Function]}
  setActiveCardIdAction={[Function]}
  type="cities"
/>
`;
